utils/progresses.js

```diff
@@ -92,16 +92,26 @@ const assertUserOrTaskExists = async (queryParams) => {
  * @param {string} queryParams.userId - (Optional) The user ID to filter progress documents by.
  * @param {string} queryParams.taskId - (Optional) The task ID to filter progress documents by.
  * @param {string} queryParams.type - (Optional) The type to filter progress documents by.
+ * @param {string} queryParams.orderBy - (Optional) The type to sort the documents.
  * @returns {Query} A Firestore query object that filters progress documents based on the given parameters.
  */
 const buildQueryToFetchDocs = (queryParams) => {
-  const { type, userId, taskId } = queryParams;
+  const { type, userId, taskId, orderBy } = queryParams;
+  const orderByField = !orderBy ? "date" : orderBy.split("-")[orderBy.split("-").length - 1];
```

Comment:
here const orderByField = !orderBy ? "date" : orderBy.split("-")[orderBy.split("-").length - 1];

This line is supposed to extract the field name from the orderBy query parameter, which is in the format field-asc/desc. However, if the orderBy parameter is not in the correct format, or if it is not provided at all, the split() function will return an empty array. In this case, the orderByField variable will be set to undefined.

This will cause an error when the query is executed, because Firestore cannot order documents by a field that does not exist.

can we think link something below ...?

const orderByField = orderBy ? orderBy.split("-")[0] : "date";

Reply:
The condition !orderBy will make sure that split will only execute for truthy i.e for string of length more than 0.

Also if a field which does not exist in any of the document is provided the firestore returns zero result not an error.

Through, the suggested change is more readable. so i have applied that